### PR TITLE
chore(targets): create page targets only when attached to them

### PR DESCRIPTION
**Pull request by yury-s**

### CODE REVIEW

File: src/chromium/crBrowser.ts [hunk 1]
```diff
@@ -102,21 +102,11 @@ export class CRBrowser extends platform.EventEmitter implements Browser {
     return createPageInNewContext(this, options);
   }
 
-  async _onAttachedToTarget(event: Protocol.Target.attachedToTargetPayload) {
-    if (!CRTarget.isPageType(event.targetInfo.type))
+  async _onAttachedToTarget({targetInfo, sessionId}: Protocol.Target.attachedToTargetPayload) {
+    if (!CRTarget.isPageType(targetInfo.type))
```
Comment (dgozman) [inline src/chromium/crBrowser.ts]: Let's assert.

File: src/chromium/crBrowser.ts [hunk 2]
```diff
@@ -135,27 +125,47 @@ export class CRBrowser extends platform.EventEmitter implements Browser {
           context.emit(Events.CRBrowserContext.BackgroundPage, event);
           break;
         }
-        case 'service_worker': {
-          const serviceWorker = await target.serviceWorker();
-          context.emit(Events.CRBrowserContext.ServiceWorker, serviceWorker);
-          break;
-        }
       }
     } catch (e) {
```
Comment (dgozman) [inline src/chromium/crBrowser.ts]: I think we can remove this catch now, since we use pageOrError().

File: src/chromium/crTarget.ts [hunk 3]
```diff
@@ -48,38 +48,32 @@ export class CRTarget {
     browser: CRBrowser,
     targetInfo: Protocol.Target.TargetInfo,
     browserContext: CRBrowserContext,
+    session: CRSession | null,
     sessionFactory: () => Promise<CRSession>) {
```
Comment (dgozman) [inline src/chromium/crTarget.ts]: Do we use sessionFactory?

**Review by dgozman — APPROVED** [review]

File: src/chromium/crBrowser.ts (hunk 2 above)
Comment (yury-s) [reply src/chromium/crBrowser.ts]: Done.

File: src/chromium/crBrowser.ts (hunk 1 above)
Comment (yury-s) [reply src/chromium/crBrowser.ts]: Done.

File: src/chromium/crTarget.ts (hunk 3 above)
Comment (yury-s) [reply src/chromium/crTarget.ts]: Yes, for service workers.

**Pull request merged**